spec.html

```diff
@@ -46724,6 +46897,83 @@ <h1>Proxy.revocable ( _target_, _handler_ )</h1>
     </emu-clause>
   </emu-clause>
 
+  <emu-clause id="sec-module-source-objects">
+    <h1>Module Source Objects</h1>
+    <p>Module Source Objects represent modules in their source import phase, which are not linked, instantiated or executed.</p>
+    <p>All Module Source Objects should have a prototype of %AbstractModuleSource%.prototype and must define a [[ModuleSourceRecord]] internal slot.</p>
+    <p>Hosts may define their own Module Source subclasses for custom module record types.</p>
```

Comment:
What does "record" mean here?

```suggestion
    <p>Hosts may define their own Module Source subclasses for custom module types.</p>
```

Also, wouldn't they be `%AbstractModuleSource%` subclasses?

```suggestion
    <p>Hosts may define their own %AbstractModuleSource% subclasses for custom module types.</p>
```

Reply:
I guess we're trying to say "Hosts may define their own %AbstractModuleSource% subclasses for custom Module Record subtypes/subclasses/subrecords."

Not sure about that last word.